orttraining/orttraining/test/python/orttraining_test_pytorch_trainer_frontend.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+import torch
+
+from onnxruntime.capi.training.optim import lr_scheduler
+from onnxruntime.capi.training.amp import loss_scaler
+from onnxruntime.capi.training import pytorch_trainer_options as pt_options
+
+
+@pytest.mark.parametrize("test_input", [
+    ({}),
+    ({'batch': {},
+      'device': {},
+      'distributed': {},
+      'mixed_precision': {},
+      'utils': {},
+      '_internal_use': {}})
+])
+def testDefaultValues(test_input):
+    ''' Test different ways of using default values for incomplete input'''
+
+    expected_values = {
+        'batch': {
+            'gradient_accumulation_steps': 0
+        },
+        'device': {
+            'id': None,
+            'mem_limit': 0
+        },
+        'distributed': {
+            'world_rank': 0,
+            'world_size': 1,
+            'local_rank': 0,
+            'allreduce_post_accumulation': False,
+            'enable_partition_optimizer': False,
+            'enable_adasum': False
+        },
+        'lr_scheduler': None,
+        'mixed_precision': {
+            'enabled': False,
+            'loss_scaler': None
+        },
+        'utils': {
+            'grad_norm_clip': False
+        },
+        '_internal_use': {
+            'frozen_weights': [],
+            'enable_internal_postprocess': True,
+            'extra_postprocess': None
+        }
+    }
+
+    actual_values = pt_options.PytorchTrainerOptions(test_input)
+    assert actual_values._validated_opts == expected_values
+
+
+def testInvalidMixedPrecisionEnabledSchema():
+    '''Test an invalid input based on schema validation error message'''
+
+    expected_msg = 'must be of boolean type'
+    actual_values = pt_options.PytorchTrainerOptions(
+        {'mixed_precision': {'enabled': 1}})
+    assert actual_values.mixed_precision[0].enabled[0] == expected_msg
```

Comment:
Is there a way to remove the use of `[0]`?

Is there a way to throw error for invalid input? It seems in this case, the response we get is that the value of options is set to be error message instead of invalid input value. I see there is a class `PytorchTrainerOptionsValidator`, can that be used to check for invalid input and throw error?

Reply:
`cerberus` validator classes doesn't throw. Instead, it sets a dictionary with all errors found in the input. This particular test has a single error with a single error message (thus [0] is used), but in real code several parameters could be malformed and cerberus provide a specific message for each

Reply:
Should we check for the error message in the constructor and throw error? I'm thinking it's not ideal that currently if fails silently (error message put in the returned `actual_values`, but users have to check manually)

Reply:
It was my intention, but I missed it :)